### PR TITLE
JDK18 adds --finalization=[enabled|disabled]

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Added `J9_EXTENDED_RUNTIME_DISABLE_FINALIZATION`;
Implemented `JVM_IsFinalizationEnabled`;
Accepted command line option `--finalization=[enabled|disabled]`.

related to https://github.com/eclipse-openj9/openj9/issues/14049 
depends on https://github.com/eclipse-openj9/openj9/pull/14113

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins compile win jdknext

**Comment by pshipton** [comment]
https://github.com/eclipse-openj9/openj9/pull/14113 requires an update so this will need to be updated as well before merge.
https://github.com/eclipse-openj9/openj9/pull/14113#issuecomment-989121701

File: runtime/vm/jvminit.c [hunk 1]
```diff
@@ -3761,6 +3761,32 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 	}
 #endif /* defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS) */
 
+#if JAVA_SPEC_VERSION >= 18
+	{
+		/* JEP 421: Deprecate Finalization for Removal
+		 * Expecting values: "enabled" and "disabled". The default value for now will be "enabled", i.e., finalization is enabled by default.
+		 * Other values will cause an error, additional values may be added in the future.
+		 */
```
Comment (keithc-ca) [inline runtime/vm/jvminit.c:3769]: Please fold this comment so it is complete visible here.

**Comment by pshipton** [comment]
Running PR build https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/1308/

File: runtime/vm/jvminit.c [hunk 2]
```diff
@@ -3761,6 +3761,32 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 	}
 #endif /* defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS) */
 
+#if JAVA_SPEC_VERSION >= 18
+	{
+		/* JEP 421: Deprecate Finalization for Removal - Expecting values: "enabled" and "disabled".
+		 * The default value for now will be "enabled", * i.e., finalization is enabled by default.
```
Comment (keithc-ca) [inline runtime/vm/jvminit.c:3767]: Please remove the extraneous `*` before 'i.e.'.

File: runtime/vm/jvminit.c (hunk 1 above)
Comment (JasonFengJ9) [reply runtime/vm/jvminit.c]: Rebased w/ https://github.com/eclipse-openj9/openj9/commit/86231f8ec44a97115ece370e70ce405e1ab4241c
Adjusted comments along with following code block.

File: runtime/vm/jvminit.c (hunk 2 above)
Comment (JasonFengJ9) [reply runtime/vm/jvminit.c]: Removed.

**Comment by pshipton** [comment]
Pls fix the conflict now.

**Comment by pshipton** [comment]
Keith's commit is merged so it can be removed from this PR.

**Comment by JasonFengJ9** [comment]
> Pls fix the conflict now.

Resolved the conflict.

**Review by pshipton — APPROVED** [review]

**Pull request merged**